src/Snowflake/Command.php

```diff
@@ -149,18 +226,270 @@ public static function cloneDatabaseFromShared(Connection $connection, array $da
+            $roles = array_merge($roles, array_combine(
+                array_map(fn($v) => $v['name'], $filteredRolesInRole),
+                $filteredRolesInRole
+            ));
+
+            $usersInRole = array_filter($ownershipToRole, fn($v) => $v['granted_on'] === 'USER');
+            $filteredUsersInRole = array_filter($usersInRole, fn($v) => !in_array($v['name'], $users));
+            $users = array_merge($users, array_map(fn($v) => $v['name'], $filteredUsersInRole));
+        }
+
+        return ['roles' => $roles, 'users' => $users];
+    }
+
+    public static function grantRoleToUser(Connection $connection, string $user, string $role): void
+    {
+        $connection->query(sprintf(
+            'GRANT ROLE %s TO USER %s',
+            QueryBuilder::quoteIdentifier($role),
+            QueryBuilder::quoteIdentifier($user)
+        ));
+    }
+
+    public static function assignGrantToRole(Connection $connection, array $grant): void
+    {
+        self::useRole($connection, $grant['granted_by']);
+
+        if ($grant['privilege'] === 'USAGE' && $grant['granted_on'] === 'ROLE') {
+            $sql = sprintf(
+                'GRANT %s %s TO %s %s %s',
+                $grant['granted_on'],
+                $grant['name'],
+                $grant['granted_to'],
+                $grant['grantee_name'],
+                $grant['grant_option'] === 'true' ? 'WITH GRANT OPTION' : '',
+            );
+        } else {
+            $sql = sprintf(
+                'GRANT %s ON %s %s TO %s %s %s',
+                $grant['privilege'],
+                $grant['granted_on'],
+                $grant['granted_on'] !== 'ACCOUNT' ? $grant['name'] : '',
+                $grant['granted_to'],
+                $grant['grantee_name'],
+                $grant['grant_option'] === 'true' ? 'WITH GRANT OPTION' : '',
+            );
+        }
+
+        $connection->query($sql);
+    }
+
+    public static function createMainRole(
+        Connection $connection,
+        string $warehouse,
+        array $users
+    ): void {
+        $mainRole = 'KEBOOLA_STORAGE';
+        $user = $mainRole;
+
+        self::createRole($connection, ['name' => $mainRole]);
+
+        $connection->query(sprintf(
+            'DROP WAREHOUSE IF EXISTS %s',
+            $warehouse
+        ));
+
+        $connection->query(sprintf(
+            'GRANT CREATE DATABASE ON ACCOUNT TO ROLE %s;',
+            $mainRole
+        ));
+
+        $connection->query(sprintf(
+            'GRANT CREATE ROLE ON ACCOUNT TO ROLE %s WITH GRANT OPTION;',
+            $mainRole
+        ));
+
+        $connection->query(sprintf(
+            'GRANT CREATE USER ON ACCOUNT TO ROLE %s WITH GRANT OPTION;',
+            $mainRole
+        ));
+
+        $createWarehouseSql = <<<SQL
+'CREATE WAREHOUSE %s WITH WAREHOUSE_SIZE = 'XSMALL' WAREHOUSE_TYPE = 'STANDARD' AUTO_SUSPEND = 300 AUTO_RESUME = TRUE;'
```

Comment:
Špatný předpoklad - musíš vytvořit všechny warehousy, které má nějaký resource nagrantovaný IMHO

Reply:
vzal jsem to z
https://keboolamanagementapi.docs.apiary.io/#reference/super-storage-backends-management/storage-backend-collection
ale jestli může mít vice warehousu tak potom ok

Reply:
jj dynamic backend size = 3 warehousy + ten defaultní

Reply:
doplnil jsem tady - https://github.com/keboola/project-migration-tool/pull/1/commits/222b620a2eb46d28b0e63e80bad66102a2413b52

zmigrují se všechny 4 nebo všechny které má grantnuté hlavní role (KEBOOLA_STORAGE)